### PR TITLE
Added docs for Models page

### PR DESCRIPTION
**Pull request by siddharthnandy**

### CODE REVIEW

File: docs/source/user-guide/models.rst [hunk 1]
```diff
@@ -1,2 +1,178 @@
...
+using the ``DownscaleLitModule`` module) and the desired model archiecture.
+Currently, three deep neural network architectures are supported:
+
+#. Convolutional neural networks: the CNN is a widely used architecture for visual recognition tasks. A constrained version of the standard neural network, CNNs capitalize on knowledge of the input's structure as an image. ClimateLearn suports two popular variants of CNNs:
+
+    #. ResNet: ResNets are a popular variant of CNNs that have been used to achieve weather forecasting for variables such as temperature and geopotential.
+
+    #. U-Net: U-Nets are a CNN varriant that entails both downsampling and upsampling convolutions. Their implementationin ClimateLearn allows users to benchmark U-Net for climate modeling tasks.
+
+#. Vision transformers: the utility of ViTs for representing climate variables is largely under-explored, so ClimateLearn provides an implementation for benchmarking transformers.
```
[the remark this answers is not in the document]
Comment (jasonjewik) [reply docs/source/user-guide/models.rst:14]: And for all three models, cite at least one paper that uses them for solving downscaling/forecasting to backup your claims.

File: docs/source/user-guide/models.rst [hunk 2]
```diff
@@ -1,2 +1,178 @@
...
+    )
+
+Training
+--------
+
+The ``climate_learn.training`` module provides a ``Trainer`` class for
+fitting and testing models. The trainer is initialized with parameters
+such as the seed, the accelerator, and the maximimum number of epochs.
+
+The module also provides a function for optionally using `Weights and
```
Comment (jasonjewik) [inline docs/source/user-guide/models.rst:50]: Leave out the wandb stuff for now. Unless you've tested and verified that it works, I wouldn't include it in the docs.

File: docs/source/user-guide/models.rst [hunk 3]
```diff
@@ -1,2 +1,178 @@
...
+    :linenos:
+
+    from climate_learn.training import Trainer, WandbLogger
+
+    trainer = Trainer(
+        seed = 0,
+        accelerator = "gpu",
+        precision = 16,
+        max_epochs = 5,
+        logger = WandbLogger(project = "climate-learn-test", name = "forecast-vit"),
```
Comment (jasonjewik) [inline docs/source/user-guide/models.rst:73]: Remove wandb.

File: docs/source/user-guide/models.rst [hunk 4]
```diff
@@ -1,2 +1,178 @@
...
+
+    from climate_learn.training import Trainer, WandbLogger
+
+    # Initialize model trainer
+    trainer = Trainer(
+        seed = 0,
+        accelerator = "gpu",
+        precision = 16,
+        max_epochs = 5,
+        # logger = WandbLogger(project = "climate-learn-test", name = "forecast-unet"),
```
Comment (jasonjewik) [inline docs/source/user-guide/models.rst:167]: Remove wandb. Also generally, idt we should leave commented-out lines of code in the docs.

File: docs/source/user-guide/models.rst (hunk 2 above)
Comment (siddharthnandy) [reply docs/source/user-guide/models.rst]: removed

File: docs/source/user-guide/models.rst (hunk 4 above)
Comment (siddharthnandy) [reply docs/source/user-guide/models.rst]: removed

File: docs/source/user-guide/models.rst (hunk 3 above)
Comment (siddharthnandy) [reply docs/source/user-guide/models.rst]: removed

File: docs/source/user-guide/models.rst [hunk 5]
```diff
@@ -1,2 +1,178 @@
...
\ No newline at end of file
+======
+ClimateLearn's model modules are configurable based on the problem
+(forecasting, using the ``ForecastLitModule`` module, and downscaling,
+using the ``DownscaleLitModule`` module) and the desired model archiecture.
+Currently, three deep neural network architectures are supported:
+
+#. Convolutional neural networks: the CNN is a widely used architecture for visual recognition tasks. A constrained version of the standard neural network, CNNs capitalize on knowledge of the input's structure as an image. ClimateLearn suports two popular variants of CNNs:
+
+    #. ResNet: ResNets are a popular variant of CNNs that have been used to achieve weather forecasting for variables such as temperature and geopotential.
```
[the remark this answers is not in the document]
Comment (siddharthnandy) [reply docs/source/user-guide/models.rst]: added links to original papers on ResNets and U-Nets

File: docs/source/user-guide/models.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (siddharthnandy) [reply docs/source/user-guide/models.rst]: added link to ViT paper

[the remark this answers is not in the document]
Comment (siddharthnandy) [reply docs/source/user-guide/models.rst]: U-Nets haven't been used for climate modeling before, only in the biomedical space, but I added links to the papers that have used ResNets and ViT's for forecasting

**Review by jasonjewik — APPROVED** [review]
lgtm

**Pull request merged**